compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -1843,17 +1844,19 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                         err.span_label(sp, format!("variant `{}` not found here", assoc_ident));
                     }
 
-                    err.emit();
+                    err.emit()
                 } else if !qself_ty.references_error() {
                     // Don't print `TyErr` to the user.
                     self.report_ambiguous_associated_type(
                         span,
                         &qself_ty.to_string(),
                         "Trait",
                         assoc_ident.name,
-                    );
-                }
-                return Err(ErrorReported);
+                    )
+                } else {
+                    ErrorReported::new_i_know_what_im_doing()
```

Comment:
The `if` this is in should be flipped, so that the `ErrorReported` can be obtained from `references_error`.